src/com/model/CashMachine.java

```diff
@@ -0,0 +1,175 @@
+     */
+    private void useBiggestBanknoteFistOperation(int sumToBePaid) {
+        int banknoteForPay = 0;
+
+        for (int leftToPay = sumToBePaid; leftToPay > 0; leftToPay -= banknoteForPay) {
+
+            for (Map.Entry<Integer, Integer> item : banknotesMap.entrySet()) {
+
+                int banknoteAmount = item.getValue();
+                int banknoteNominal = item.getKey();
+
+                if (banknoteAmount > 0 && (banknoteNominal > banknoteForPay || banknoteNominal == leftToPay)) {
+                    banknoteForPay = banknoteNominal;
+
+                    if (banknoteForPay == leftToPay) {
+                        break;
+                    }
+                }
+            }
+
+            // If you need to add a smaller denomination, to pay out the desired amount.
+            if (banknoteForPay > leftToPay) {
+                banknoteForPay = findNearestBanknote(leftToPay);
+            }
+
+            banknotesCounter.decrementBanknoteCounter(banknoteForPay);
+            banknotesHistory.banknotePayoutWriteInHistory(banknoteForPay);
+        }
+    }
+
+    /**
+     * Method will be in class Strategy in homework_2
+     * Implemented the second strategy;
+     *
+     * @param sumToBePaid - Get amount of needed money from user
+     */
+    private void useBanknotesEvenlyStrategy(int sumToBePaid) {
+        int banknoteForPay = 0;
+        int leftToPay;
+
+        for (leftToPay = sumToBePaid; leftToPay > 0; leftToPay -= banknoteForPay) {
+            int biggestAmount = 0;
+
+            for (Map.Entry<Integer, Integer> item : banknotesMap.entrySet()) {
+
+                int banknoteAmount = item.getValue();
+                if (banknoteAmount > 0 && banknoteAmount > biggestAmount) {
+                    biggestAmount = banknoteAmount;
+                    banknoteForPay = item.getKey();
+                }
+            }
+
+            // If you need to add a smaller denomination, to pay out the desired amount.
+            if (banknoteForPay > leftToPay) {
+                banknoteForPay = findNearestBanknote(leftToPay);
+            }
+
+            banknotesCounter.decrementBanknoteCounter(banknoteForPay);
+            banknotesHistory.banknotePayoutWriteInHistory(banknoteForPay);
+        }
+    }
+
+    /**
+     * If there is no such currency in {@link #banknotesMap} or the currency is not available -
+     * add + 1 and repeat the loop;
+     *
+     * @param leftToPay - How much more is needed to pay the full amount;
+     * @return The desired denomination of the banknote or the closest one in a big way;
+     */
+    private int findNearestBanknote(int leftToPay) {
+        int banknoteAmount;
+        int banknoteNominal = leftToPay;
+        if (banknotesMap.containsKey(banknoteNominal)) {
+            banknoteAmount = banknotesMap.get(banknoteNominal);
+
+            if (banknoteAmount > 0) {
+                return banknoteNominal;
+            } else {
```

Comment:
`else` is redundant here. Technically, `else` is needed only when this is a "temporary alternative" code branch after which code executions will go on. But in this case, we have the `return` in line 166. Therefore, either we are here, or the execution has already finished, and we are not going any further, anyway.
In such cases we write
```
if (something) {
   return foo;
}
return bar;
```
Generally speaking, the fewer levels of nestedness you use (I mean, curly braces, and braces in braces, etc.) -- the better)